### PR TITLE
codeintel: extract and decouple ranking background jobs from service

### PR DESCRIPTION
**Pull request by efritz**
Do the background thing.

## Test plan

Ran locally.

### CODE REVIEW

File: enterprise/internal/codeintel/ranking/internal/background/loader_metrics.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+func newLoaderMetrics(observationContext *observation.Context) *loaderMetrics {
+	counter := func(name, help string) prometheus.Counter {
+		counter := prometheus.NewCounter(prometheus.CounterOpts{
+			Name: name,
+			Help: help,
+		})
+
+		fmt.Printf("!!!!!!!!LOADING %s\n", name)
```
Comment (Strum355) [inline enterprise/internal/codeintel/ranking/internal/background/loader_metrics.go:23]: epic

Comment (efritz) [reply enterprise/internal/codeintel/ranking/internal/background/loader_metrics.go]: Fixed

File: enterprise/internal/codeintel/ranking/init.go [hunk 2]
```diff
@@ -91,3 +94,42 @@ var initServiceMemo = memo.NewMemoizedConstructorWithArg(func(deps serviceDepend
 func scopedContext(component string) *observation.Context {
 	return observation.ScopedContext("codeintel", "ranking", component)
 }
+
+func NewIndexer(service *Service, observationContext *observation.Context) []goroutine.BackgroundRoutine {
+	return []goroutine.BackgroundRoutine{
+		background.NewRepositoryIndexer(
+			service.store,
+			service.gitserverClient,
+			service.symbolsClient,
```
Comment (Strum355) [inline enterprise/internal/codeintel/ranking/init.go:103]: I personally pass these things in independently, dont need to instantiate a service if we're not using it. Minor nit though, so w/e

Comment (Strum355) [reply enterprise/internal/codeintel/ranking/init.go:103]: Ah im assuming this is related to the stores being internal?

**Review by Strum355 — APPROVED** [review]

**Pull request merged**